server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,413 @@
+                @Override
+                public void onResponse(Void unused) {
+                    logger.info(
+                        "successfully removed decommissioned cluster manager eligible nodes [{}] from voting config ",
+                        clusterManagerNodesToBeDecommissioned.toString()
+                    );
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.debug(
+                        new ParameterizedMessage("failure in removing decommissioned cluster manager eligible nodes from voting config"),
+                        e
+                    );
+                }
+            }
+        );
+    }
+
+    /**
+     * Registers new decommissioned attribute metadata in the cluster state with {@link DecommissionStatus#DECOMMISSION_INIT}
+     * <p>
+     * This method can be only called on the cluster-manager node. It tries to create a new decommissioned attribute on the cluster manager
+     * and if it was successful it adds new decommissioned attribute to cluster metadata.
+     * <p>
+     * This method would only be executed on eligible cluster manager node
+     *
+     * @param decommissionAttribute register decommission attribute in the metadata request
+     * @param listener              register decommission listener
+     */
+    private void registerDecommissionAttribute(
+        final DecommissionAttribute decommissionAttribute,
+        final ActionListener<ClusterStateUpdateResponse> listener
+    ) {
+        clusterService.submitStateUpdateTask(
+            "put_decommission [" + decommissionAttribute + "]",
+            new ClusterStateUpdateTask(Priority.URGENT) {
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    Metadata metadata = currentState.metadata();
+                    Metadata.Builder mdBuilder = Metadata.builder(metadata);
+                    DecommissionAttributeMetadata decommissionAttributeMetadata = metadata.custom(DecommissionAttributeMetadata.TYPE);
+                    ensureNoAwarenessAttributeDecommissioned(decommissionAttributeMetadata, decommissionAttribute);
+                    decommissionAttributeMetadata = new DecommissionAttributeMetadata(decommissionAttribute);
+                    mdBuilder.putCustom(DecommissionAttributeMetadata.TYPE, decommissionAttributeMetadata);
+                    logger.info(
+                        "registering decommission metadata for attribute [{}] with status as [{}]",
+                        decommissionAttribute.toString(),
+                        DecommissionStatus.DECOMMISSION_INIT
+                    );
+                    return ClusterState.builder(currentState).metadata(mdBuilder).build();
+                }
+
+                @Override
+                public void onFailure(String source, Exception e) {
+                    if (e instanceof DecommissioningFailedException) {
```

Comment:
Here, before we execute the cluster state update we are again checking if there's no inflight request getting executed. If yes, we are throwing the exception. And hence in that case, it might land up here

Reply:
Update - Logic changed a bit. In case if there's any validation failures etc, we directly call listener.onFailure(). No separate handling for different exceptions that we might get